### PR TITLE
Feat add board button

### PR DESCRIPTION
**Pull request by TomSpencerLondon**
Ticket: https://trello.com/c/rdMNF064

### CODE REVIEW

File: src/api/__mocks__/boardsApi.js [hunk 1]
```diff
@@ -7,3 +7,5 @@ export const getBoard = async () => (getFail() ? Promise.reject(new Error('fail'
 export const getBoards = async () => (getFail()
   ? Promise.reject(new Error('fail'))
   : new Promise((resolve) => resolve(getMockResponse())));
+
+export const saveBoard = async () => (new Promise((resolve) => resolve(getMockResponse())));
```
Comment (Morinslash) [inline src/api/__mocks__/boardsApi.js]: this file can be refactor to remove code duplication

File: src/api/boardsApi.js [hunk 2]
```diff
@@ -1,4 +1,18 @@
...
+}
+
+async function invalidError() {
+  throw new Error('Invalid board request');
+}
+
+export const saveBoard = (board) => {
+  if (invalid(board)) return invalidError();
+
+  return fetchWrapper({ endpoint: '/boards', method: 'POST', body: board });
```
Comment (Morinslash) [inline src/api/boardsApi.js]: please, extract '/boards' to constant to remove string duplication

File: src/api/boardsApi.test.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    });
+  });
+});
+
+describe('getBoards', () => {
+  it('calls fetchWrapper with /boards endpoint', async () => {
+    fetchApiSpy = jest.spyOn(fetchApi, 'fetchWrapper').mockImplementationOnce(() => {});
+    await getBoards();
+    expect(fetchApiSpy).toHaveBeenCalledWith({
+      endpoint: '/boards'
```
Comment (Morinslash) [inline src/api/boardsApi.test.js]: same as before, we can extract /boards to constant in file to remove duplication

File: src/api/cardsApi.test.js [hunk 4]
```diff
@@ -1,42 +1,151 @@
...
+    });
+  });
+
+  describe('valid request', () => {
+    it('calls fetchWrapper with updated text', async () => {
+      fetchApiSpy = jest.spyOn(fetchApi, 'fetchWrapper').mockImplementationOnce(() => {});
+      const newText = 'text';
+      await editCard(1, newText);
+      expect(fetchApiSpy).toHaveBeenCalledWith({
+        endpoint: '/cards/1',
```
Comment (Morinslash) [inline src/api/cardsApi.test.js]: please change URL to not be hard coded cardID, then we can extract /cards to constant for all test cases in file

File: src/components/AddBoardModal/AddBoardModal.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    e.preventDefault();
+    saveBoard({ title: text, userEmail: getUserEmail() }).then((newBoard) => {
+      setError(false);
+      handleClose();
+      history.push(`/${newBoard.id}`);
+    }).catch(() => setError(true));
+  };
+
+  return (
+    <div>
```
Comment (Morinslash) [inline src/components/AddBoardModal/AddBoardModal.js]: please use ReactFragment instead of div

**Review by Morinslash — CHANGES_REQUESTED** [review]
Need some small refactor only, good job!

File: src/api/__mocks__/boardsApi.js (hunk 1 above)
Comment (Morinslash) [reply src/api/__mocks__/boardsApi.js]:
```
const { getFail, getMockResponse } = require('./mockHelper');
const dummyData = require('./boardsApiMockResponse.json');

function mockedResponse() {
  return getFail()
    ? Promise.reject(new Error('fail'))
    : new Promise((resolve) => resolve(getMockResponse()));
}

export const getBoard = async () => (getFail() ? Promise.reject(new Error('fail'))
  : new Promise((resolve) => resolve(dummyData)));

export const getBoards = async () => mockedResponse();

export const saveBoard = async () => mockedResponse();
```

**Review by Morinslash — APPROVED** [review]

File: src/api/__mocks__/mockHelper.js [hunk 6]
```diff
@@ -16,3 +16,9 @@ export function setMockResponse(data) {
 export function getMockResponse() {
```
Comment (ewansheldon) [inline src/api/__mocks__/mockHelper.js]: i don't think this need to be exported anymore as it's just used within this module

File: src/components/Home/Home.js [hunk 7]
```diff
@@ -42,21 +41,28 @@ const Home = ({ history }) => {
...
-            </Card>
-          </Link>
-        </GridListTile>
-      ))}
-    </GridList>
+    <>
+      <Grid container spacing={3}>
+        <Grid item xs={12} sm={3} key={0}>
+          <Grid container justify="center">
+            <AddBoardModal history={history} />
```
Comment (ewansheldon) [inline src/components/Home/Home.js:48]: this button won't render if you have no boards

**Review by ewansheldon — CHANGES_REQUESTED** [review]

**Review by ewansheldon — APPROVED** [review]

**Pull request merged**